packages/amplify_datastore_plugin_interface/lib/src/types/models/model_field_type.dart

```diff
@@ -20,12 +20,16 @@ class ModelFieldType {
 
   final String? ofModelName; //opt
 
-  const ModelFieldType(this.fieldType, {this.ofModelName});
+  final String? ofCustomTypeName;
+
+  const ModelFieldType(this.fieldType,
+      {this.ofModelName, this.ofCustomTypeName});
 
   Map<String, dynamic> toMap() {
     Map<String, dynamic> map = {
       'fieldType': describeEnum(fieldType),
       'ofModelName': ofModelName,
+      'ofCustomTypeName': ofCustomTypeName,
     };
     return Map.from(map)..removeWhere((k, v) => v == null);
```

Comment:
Nit: This might be a cleaner pattern
```suggestion
      'fieldType': describeEnum(fieldType),
      if (ofModelName != null) 'ofModelName': ofModelName,
      if (ofCustomTypeName != null) 'ofCustomTypeName': ofCustomTypeName,
    };
    return map;
```